magma/bits.py

```diff
@@ -487,13 +487,13 @@ def get_family(cls):
         return m.get_family()
 
     def unused(self):
-        if not self.is_output():
-            raise TypeError("unused can only be used on output")
+        if self.is_input() or self.is_inout():
+            raise TypeError("unused cannot be used with input/inout")
         m.wire(self, DefineUnused(len(self))().I)
 
     def undriven(self):
-        if not self.is_input():
-            raise TypeError("undriven can only be used on input")
+        if self.is_output() or self.is_inout():
+            raise TypeError("undriven cannot be used with output/inout")
```

Comment:
These type checks needed to be updated because they don't properly handle temporary values, which are neither inputs/outputs/inouts